Test failure: AdvancedNetworkingClientCommunicationIntegrationTest.testClientConnectionToEndpoints

Failure in: http://jenkins.hazelcast.com/view/Hazelcast/job/Hazelcast-4.master-nightly/759/consoleFull


Related test output: 
```
22:37:58 [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 303.352 s <<< FAILURE! - in com.hazelcast.client.impl.connection.tcp.AdvancedNetworkingClientCommunicationIntegrationTest
22:37:58 [ERROR] testClientConnectionToEndpoints(com.hazelcast.client.impl.connection.tcp.AdvancedNetworkingClientCommunicationIntegrationTest)  Time elapsed: 303.352 s  <<< ERROR!
22:37:58 org.junit.runners.model.TestTimedOutException: test timed out after 300000 milliseconds
22:37:58 	at java.lang.Thread.sleep(Native Method)
22:37:58 	at com.hazelcast.client.impl.connection.tcp.WaitStrategy.sleep(WaitStrategy.java:75)
22:37:58 	at com.hazelcast.client.impl.connection.tcp.TcpClientConnectionManager.doConnectToCandidateCluster(TcpClientConnectionManager.java:501)
22:37:58 	at com.hazelcast.client.impl.connection.tcp.TcpClientConnectionManager.doConnectToCluster(TcpClientConnectionManager.java:406)
22:37:58 	at com.hazelcast.client.impl.connection.tcp.TcpClientConnectionManager.connectToCluster(TcpClientConnectionManager.java:367)
22:37:58 	at com.hazelcast.client.impl.connection.tcp.TcpClientConnectionManager.start(TcpClientConnectionManager.java:316)
22:37:58 	at com.hazelcast.client.impl.clientside.HazelcastClientInstanceImpl.start(HazelcastClientInstanceImpl.java:369)
22:37:58 	at com.hazelcast.client.HazelcastClient.constructHazelcastClient(HazelcastClient.java:460)
22:37:58 	at com.hazelcast.client.HazelcastClient.newHazelcastClientInternal(HazelcastClient.java:416)
22:37:58 	at com.hazelcast.client.HazelcastClient.newHazelcastClient(HazelcastClient.java:136)
22:37:58 	at com.hazelcast.client.impl.connection.tcp.AdvancedNetworkingClientCommunicationIntegrationTest.createClient(AdvancedNetworkingClientCommunicationIntegrationTest.java:61)
22:37:58 	at com.hazelcast.client.impl.connection.tcp.AdvancedNetworkingClientCommunicationIntegrationTest.testClientConnectFailOnPort(AdvancedNetworkingClientCommunicationIntegrationTest.java:66)
22:37:58 	at com.hazelcast.client.impl.connection.tcp.AdvancedNetworkingClientCommunicationIntegrationTest.testClientConnectionToEndpoints(AdvancedNetworkingClientCommunicationIntegrationTest.java:51)
22:37:58 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
22:37:58 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
22:37:58 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
22:37:58 	at java.lang.reflect.Method.invoke(Method.java:498)
22:37:58 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
22:37:58 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
22:37:58 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
22:37:58 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
22:37:58 	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
22:37:58 	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
22:37:58 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
22:37:58 	at java.lang.Thread.run(Thread.java:748)
22:37:58 
```

--- a/hazelcast/src/test/java/com/hazelcast/client/impl/connection/tcp/AdvancedNetworkingClientCommunicationIntegrationTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/client/impl/connection/tcp/AdvancedNetworkingClientCommunicationIntegrationTest.java
@@ -58,6 +58,7 @@ public class AdvancedNetworkingClientCommunicationIntegrationTest extends Abstra
         ClientConfig clientConfig = new ClientConfig();
         clientConfig.getNetworkConfig().addAddress("127.0.0.1:" + port);
         clientConfig.getNetworkConfig().setConnectionTimeout(3000);
+        clientConfig.getConnectionStrategyConfig().getConnectionRetryConfig().setClusterConnectTimeoutMillis(3000);
         return HazelcastClient.newHazelcastClient(clientConfig);
     }
 

